qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+def set_hpt(vm, test, hpt_size):
+    """Run this step to set HPT in guest"""
+    session = vm.wait_for_login(timeout=600)
+    set_cmd = "echo %s > /sys/kernel/debug/powerpc/hpt_order"
+    try:
+        st, o = session.cmd_status_output(set_cmd % (hpt_size), timeout=60)
+        n_output = str(o)
+    except Exception as detail:
+        if st == 1:
+            # valid parameters,hpt_size > = 18 and hpt_size < = 46
+            if hpt_size >= 18 and hpt_size <= 46:
+                if re.search(r"Operation not permitted", n_output):
+                    pass
+                elif re.rearch(r"No space left on device", n_output):
+                    pass
+                else:
+                    test.fail("Valid param but: '%s'" % n_output)
+            # invalid parameter 0,it is handled in different way
+            # by kernel so dealing with it particularly
+            elif hpt_size == 0:
+                if not re.search(r"Input/output error", n_output):
+                    test.fail("Invalid param zero : '%s'" % n_output)
+            # invalid parameters,hpt_size < 18 and hpt_size > 46
+            else:
+                if not re.search(r"Invalid argument", n_output):
+                    test.fail("Invalid param: '%s'" % n_output)
+
+
+def get_hpt(test, params, session, hpt_size, get_cmd):
```

Comment:
Why not keep the `get_cmd` in this function instead?

Reply:
Change it to this function.